components/match2/wikis/trackmania/match_group_input_custom.lua

```diff
@@ -0,0 +1,462 @@
+		local parent = Variables.varDefault(
+			'tournament_parent',
+			mw.title.getCurrentTitle().text
+		)
+		local pageName = mw.ext.TeamLiquidIntegration.resolve_redirect(name)
+		local data = mw.ext.LiquipediaDB.lpdb('broadcasters', {
+			conditions = '[[page::' .. pageName .. ']] AND [[parent::' .. parent .. ']]',
+			query = 'flag, id',
+			limit = 1,
+		})
+		if type(data) == 'table' and data[1] then
+			flag = String.isNotEmpty(flag) and flag or data[1].flag
+			displayName = String.isNotEmpty(displayName) and displayName or data[1].id
+		end
+	end
+	if String.isNotEmpty(flag) then
+		Variables.varDefine(name .. '_flag', flag)
+	end
+	if String.isEmpty(displayName) then
+		displayName = name
+	end
+	if String.isNotEmpty(displayName) then
+		Variables.varDefine(name .. '_dn', displayName)
+	end
+	return {
+		name = name,
+		displayName = displayName,
+		flag = flag,
+	}
+end
+
+function matchFunctions.getOpponents(args)
+	-- read opponents and ignore empty ones
+	local opponents = {}
+	local isScoreSet = false
+	for opponentIndex = 1, MAX_NUM_OPPONENTS do
+		-- read opponent
+		local opponent = args['opponent' .. opponentIndex]
+		if not Logic.isEmpty(opponent) then
+			CustomMatchGroupInput.processOpponent(opponent, args.date)
+
+			-- Retrieve icon for team
+			if opponent.type == Opponent.team then
+				opponent.icon, opponent.icondark = opponentFunctions.getTeamIcon(opponent.template)
+			end
+
+			-- apply status
+			if TypeUtil.isNumeric(opponent.score) then
+				opponent.status = STATUS_HAS_SCORE
+				isScoreSet = true
+			elseif Table.includes(ALLOWED_STATUSES, opponent.score) then
+				opponent.status = opponent.score
+				opponent.score = -1
+			end
+
+			--set Walkover from Opponent status
+			args.walkover = args.walkover or STATUS_TO_WALKOVER[opponent.status]
+
+			opponents[opponentIndex] = opponent
+
+			-- get players from vars for teams
+			if opponent.type == 'team' and not Logic.isEmpty(opponent.name) then
+				args = matchFunctions.getPlayers(args, opponentIndex, opponent.name)
+			end
+		end
+	end
+
+	--set resulttype to 'default' if walkover is set
+	if args.walkover then
+		args.resulttype = 'default'
+	end
+
+	local autofinished = String.isNotEmpty(args.autofinished) and args.autofinished or true
+	-- see if match should actually be finished if score is set
+	if isScoreSet and Logic.readBool(autofinished) and not Logic.readBool(args.finished) then
+		local currentUnixTime = os.time(os.date('!*t'))
+		local lang = mw.getContentLanguage()
+		local matchUnixTime = tonumber(lang:formatDate('U', args.date))
+		local threshold = args.dateexact and 30800 or 86400
+		if matchUnixTime + threshold < currentUnixTime then
+			args.finished = true
+		end
+	end
+
+	-- apply placements and winner if finshed
+	local winner = tostring(args.winner or '')
+	if Logic.readBool(args.finished) then
+		local placement = 1
+		local lastScore
+		local lastPlacement = 1
+		local lastStatus
+		-- luacheck: push ignore
```

Comment:
not a super fan of having to trick the linter
maybe a rewrite of the stuff that gets ignored by the linter can resolve it

Reply:
do I remove this comment line and we see what's the linter is saying ?

Reply:
yeah you can try that

Reply:
https://github.com/Liquipedia/Lua-Modules/pull/2273/commits/ddc7d5f651cd8db98a35e8a19f479b3c233eef53

Reply:
So now we can try to see what it says

Reply:
Seems like nothing is triggering the linter

Reply:
nice